lib/ecco/row_type.rb

```diff
@@ -0,0 +1,40 @@
+module Ecco
+  class RowType
+    java_import com.github.shyiko.mysql.binlog.event.EventType
+
+    # MySQL v1 and v2 row events
+    EVENTS = {
+      EventType::WRITE_ROWS      => "WRITE_ROWS",
+      EventType::EXT_WRITE_ROWS  => "WRITE_ROWS",
+      EventType::UPDATE_ROWS     => "UPDATE_ROWS",
+      EventType::EXT_UPDATE_ROWS => "UPDATE_ROWS",
+      EventType::DELETE_ROWS     => "DELETE_ROWS",
+      EventType::EXT_DELETE_ROWS => "DELETE_ROWS"
+    }
+
+    TABLE_EVENT = EventType::TABLE_MAP
+
+    def self.accepted_events
+      EVENTS.keys
+    end
+
+    def self.fetch_event(type)
+      EVENTS.fetch(type)
+    end
+  end
+
+  class RowTypeSave < RowType
+    SAVE_EVENTS = {
```

Comment:
Seeing it's only the constant that's different between this class and its parent, I have a hunch things could be simplified, I don't think one needs to implement `.accepted_events` and `.fetch_event` here again if refactored a bit

Reply:
Hmm I think I know what to do.

Reply:
I'm investigating if this is a viable solution right now, it is currently not going through the spec though:

```
module Ecco
  class RowType
    java_import com.github.shyiko.mysql.binlog.event.EventType

    # MySQL v1 and v2 row events
    include_events = {
      EventType::WRITE_ROWS      => "WRITE_ROWS",
      EventType::EXT_WRITE_ROWS  => "WRITE_ROWS",
      EventType::UPDATE_ROWS     => "UPDATE_ROWS",
      EventType::EXT_UPDATE_ROWS => "UPDATE_ROWS",
      EventType::DELETE_ROWS     => "DELETE_ROWS",
      EventType::EXT_DELETE_ROWS => "DELETE_ROWS"
    }

    EVENTS = Hash.new.merge(include_events)

    TABLE_EVENT = EventType::TABLE_MAP

    def self.accepted_events
      EVENTS.keys
    end

    def self.fetch_event(type)
      EVENTS.fetch(type)
    end
  end

  class RowTypeSave < RowType
    include_events = {
      EventType::QUERY  => "QUERY",
      EventType::ROTATE => "ROTATE",
    }.merge(EVENTS)
  end
end
```

Reply:
I think you can override constans without warnings, this is OK:

```
class Parent
  CONSTANT = 1
end

class Child < Parent
  CONSTANT = 2
end

puts Parent::CONSTANT
=> 1

puts Child::CONSTANT
=> 2
```

Reply:
Just tried this in pry, you're right! That makes things easier! :)

Reply:
Now the child class only overrides the constant! 267f14e57d05f7f366f6b1b952aa0f013e984ad5